### PR TITLE
Okta Verify Authenticator Guide for Javascript SDK

### PR DESCRIPTION
**Pull request by robertnicolo-okta**
## Description:

Okta Verify Authenticator Guide for the Javascript SDK

### Resolves:

* [OKTA-459396](https://oktainc.atlassian.net/browse/OKTA-459396)

### Latest Deploy link (updated 2022-02-28):

https://621e79bcb865ef09bec93ebf--reverent-murdock-829d24.netlify.app/docs/guides/authenticators-okta-verify/nodeexpress/main/

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+---
+
+<div class="oie-embedded-sdk">
+
+<ApiLifecycle access="ie" /><br>
+<ApiLifecycle access="Limited GA" /><br>
+
+<StackSelector />
+
+This guide shows you how to integrate Okta Verify into your app using the embedded SDK.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:12]: This text is almost a repetition of the title... why is it here?

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+---
+**Learning outcomes**
+
+* Understand the Okta Verify enrollment and challenge flows
+* Learn step-by-step how to integrate Okta Verify into your authentication use case
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:18]: "Learn" isn't required as this is the "Learning outcomes" section. Assuming it's about integration the use case (singular) of Okta Verify, then how about:

Integrating authentication using Okta Verify (into your ???)

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+...
+
+<!--
+<div class="common-image-format">
+
+![Diagram showing the WebAuthn enrollment flow](/img/authenticators/authenticators-webauthn-flow-overview.png)
+
+</div>
+
+As the service provider, you can provide WebAuthn support to your users by enabling it in your Okta org and building out support for it in your application using the Embedding SDK.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:46]: Is the subject of this sentence WebAuthn support or the service provider? What is "service provider" in this context... a set of APIs on a server that's run by the organization? What is the "application" and how does that related to "service provider"? How does WebAuthn related to Okta Verify... does Okta Verify implement it?

I think that it may be something like:
Provide WebAuthn support in your application first by enabling Okta Verify in your Okta org, and then by using the Embedded SDK to add the functionality to your app.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+-->
+
+## Update configurations
+<!--
+
+Before you can start using Web Authentication (WebAuthn), create an Okta org application as described in <StackSnippet snippet="orgconfigurepwdonly" inline/>. Then add WebAuthn to your app integration by executing the following steps:
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:52]: - The title "Update configurations" isn't related to the introductory above. The current text is "enabling it in your Okta org", not "Update the configurations of", or "enable it... by updating the configuration of..." There's no bridge to help the reader understand why this section is relevant.
- I think the first sentence is trying to make sure the developer already has an application integration? If so, we should be consistent in the use of the term. 
- If sentence 1 is about ensuring there's an app integration, then the text is leading with the least important part... adding to the app integration is the primary concept. Updating the integration for WebAuthn is the most important thing.

Add WebAuthn to your Okta org Application Integration by following the steps below. For information on creating an Application Integration, see <StackSnippet snippet="orgconfigurepwdonly" inline/>.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+### Add WebAuthn to the Okta org
+
+1. In the Admin Console, go to **Security > Authenticators**.
+1. On the **Authenticators** page, click **Add Authenticator**.
+1. On the **Add Authenticator** dialog, click **Add** under **FIDO2 (WebAuthn)**.
+1. Leave the default value for **User Verification**, which is set to "Discouraged".
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:59]: What if the default value changes in the future? Can we maintenance proof the article? Also, is this a step or an extension of the previous step? How will the developer encounter this (for example, do they click Add which then brings up some dialog? a form? opens the configuration panel for the integration...)

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+-->
+
+## Update configurations
+<!--
+
+Before you can start using Web Authentication (WebAuthn), create an Okta org application as described in <StackSnippet snippet="orgconfigurepwdonly" inline/>. Then add WebAuthn to your app integration by executing the following steps:
+
+### Add WebAuthn to the Okta org
+
+1. In the Admin Console, go to **Security > Authenticators**.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:56]: This seems like 2 steps:
1. Launch the Admin Console for your Okta org.
1. Chose **Security > Authenticators**.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+1. On the **Authenticators** page, click **Add Authenticator**.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:57]: Aren't they already on the Authenticators page? Why not just say "Click **Add Authenticator**. If the developer needs orienting then you could modify the last step to be:
1. Choose **Security > Authenticators**. The Authenticators page opens and displays the Setup panel.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+1. On the **Add Authenticator** dialog, click **Add** under **FIDO2 (WebAuthn)**.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:58]: Same comment as above, they know they are on that dialog (and is that "on" or "in"). Also, assuming the final UI is the tile-based noisy one I'm seeing on the current Okta Developer preview org, I'd suggest rewording the text that orients them to the correct tile. Perhaps

1. Locate the authentication method with the title **FIDO2 (WebAuthN)** and click the Add button in that tile (or box or whatever term we use).

Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:58]: Now that I'm actually doing the flow, I think this also needs to add a second sentence:
The **Add FIDO2 (WebAuthn)** configuration dialog opens.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md (hunk 5 above)
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:59]: Per the going through the flow, I'd combine this step and the next into something like:
- Verify that **User Verification** is set to "Discouraged" and then click **Add**.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+1. Leave the default value for **User Verification**, which is set to "Discouraged".
+1. On the **Add FIDO2(WebAuthn)** dialog, click **Add**.
+1. On the **Authenticators** page, select the **Enrollment** tab.
+1. On the **Enrollment** tab, click **Edit** for the **Default Policy**.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:62]: These two steps seem like one. Are we expecting developers to stop during this flow and then to come back to it? If not, why do we add text that tells them where there already are? In this case they've just clicked Add, the dialog has gone away, and they're back on the Authenticators page. Unless they go do some other task for a while and then come back at this step, all the text "On the **Authenticators** page**" does is force their brain to process a piece of information with no value.

These two steps could be combined to:
Select the Enrollment tab, and click Edit for the Default Policy.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+1. Set **FIDO2 (WebAuthn)** to **Optional** and click **Update Policy**.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:63]: FYI, It's optional by default, at least in this implementation.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+### Configure your Okta org application to use WebAuthn
+
+1. In the Admin Console, go to **Applications** and **Applications**.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:67]: A couple of things. This is inconsistent with the instruction to open the Authenticators page. That one treats the open like a menu (which is what I thought it was.) In that case, this one should be "choose Applications > Applications."

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+1. On the **Applications** page, click on the application you've previously created.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:68]: Same comment on Application page... it's extra cognitive load. Also, I don't think "previously created" is specific enough... they haven't created one in this guide, so which one? And what are they clicking on? If they click the wrong thing they'll get the drop down to assign users, groups, or deactivate the app. How about:

"Click on the name of the desired application integration."

That makes the click target clear, though it may not be specific enough on which application... it could be more specific:

"Click the name of the application integration to which you'll add WebAuthn" or something similar.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+1. On the **General** tab ensure that **Interaction Code** and **Refresh Token** are selected.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:69]: Is that specific enough? It's not just the General tab, it's the General Settings pane of that tab, and it requires editing that pane which is somewhat unintuitive. The simplest thing in cases like this is to tell the developer to select those items... that way if the items are unchecked they get checked. That would be something like:

1. Click **Edit** in the **General Settings** pane.
1. Select **Interaction Code** and **Refresh Token** in the **Application** section.
1. Click **Save** at the bottom of the **General Settings** section.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+1. On the **Sign-On** tab, scroll down to the **Sign-On Policy** section and click **Add Rule**.
+1. On the **Add Rule** dialog box, do the following:
+   1. Enter a name for the new rule (for example "2FA Rule").
+   1. Set **User must authenticate with** to **Password+Another Factor**.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:73]: This appears to be either missing a step or a location specifier. That item is in the "Then" area of the rule. Perhaps
1. Set **User must authenticate with** to **Password+Another Factor** in the **Then** section of the rule.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+   1. Select **Device Bound**.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:74]: Isn't this an "and then" on the previous step?

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+   1. Confirm **Your org's authenticators that satisify this requirment** is set to **Password AND FIDO2 (WebAuthn) or ...**.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:75]: First off, typos for "satisify" and "requirment".
This seems a bit complex as a description. Perhaps something like:
1. Confirm that the authenticators shown in the setting **Your org's authenticators that satisfy this requirement** include both **Password** and **FIDO2 (WebAuthn)**.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+
+1. In the Admin Console, go to **Applications** and **Applications**.
+1. On the **Applications** page, click on the application you've previously created.
+1. On the **General** tab ensure that **Interaction Code** and **Refresh Token** are selected.
+1. On the **Sign-On** tab, scroll down to the **Sign-On Policy** section and click **Add Rule**.
+1. On the **Add Rule** dialog box, do the following:
+   1. Enter a name for the new rule (for example "2FA Rule").
+   1. Set **User must authenticate with** to **Password+Another Factor**.
+   1. Select **Device Bound**.
+   1. Confirm **Your org's authenticators that satisify this requirment** is set to **Password AND Okta Verify or ...**.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:87]: Location isn't well specified for two reasons. First, the information isn't presented as a single line joined with an AND, instead it's spread over 4 lines. And second they are now searching for a title of a box rather than text next to a checkbox or radio button (BTW, from a UI perspective, it's totally unintuitive as to how the two radio buttons and the box for "ANDAccess with Okta FastPass is granted" are related... though that's not our problem ;-)). Perhaps something like:

Confirm that the conditions in the ** Your org’s authenticators that satisfy this requirement** box of the **Access with Okta FastPass is granted** setting include both **Password** and **Okta Verify or FIDO2 (WebAuthn)**.


Confirm that the conditions in Your org’s authenticators that satisfy this requirement

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+   1. Click **Save** to save your changes.
+
+### Update Okta Verify to use push notification number challenge
+
+The embedded SDKs currently don't support using the push notification number challenge. If you need to use this option use the embedded widget or redirect option.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:92]: "If you need to use" isn't really needed. Also, should there be links to each of the different options (embedded widget and redirect). Also, is the subject the "embedded SDKs" or the "number notification challenge"? I think it may be:
The push notification number challenge is currently unsupported by the embedded SDKs. Use the embedded widget or redirect options to implement this method/challenge/???.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md (hunk 1 above)
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/index.md:12]: Assuming it's an abstract of the article for SSO and for the developer to figure out if it's something they need, how about:

Integrate Okta Verify into your app using the embedded SDK.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/nodeexpress/enrollmentqrcodeintegrationsteps.md [hunk 19]
```diff
@@ -0,0 +1,90 @@
+### 1: Initiate the sign-in and choose Okta Verify as the authenticator
+
+First, the user initiates the user sign-in with username and password and chooses Okta Verify as the authenticator to enroll in. These steps are common between enrollment flows and are described in [Initiate sign in for Okta Verify enrollment](#initiate-sign-in-for-okta-verify-enrollment).
+
+### 2: Display QR Code
+
+After the user chooses Okta Verify (which is the last step in [Initiate sign in for Okta Verify enrollment](#initiate-sign-in-for-okta-verify-enrollment)), display the QR Code using the `IdxTransaction` response from `OktaAuth.idx.proceed()`.  In future steps, the user will scan this QR Code on their mobile device using Okta Verify.
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/nodeexpress/enrollmentqrcodeintegrationsteps.md:7]:
```suggestion
Display the QR Code using the `IdxTransaction` response from `OktaAuth.idx.proceed()`.  In future steps, the user will scan this QR Code on their mobile device using Okta Verify.
```

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/nodeexpress/enrollmentqrcodeintegrationsteps.md [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+
+![Okta Verify option shown in UI](/img/authenticators/authenticators-oktaverify-enroll-qr-code.png)
+
+</div>
+
+### 3: Poll SDK while enrollment is ongoing
+
+`IdxTransaction.status` of `PENDING` and `nextStep.name` equal to `enroll-poll`, indicates that the user needs to complete the challenge in Okta Verify, and your application should begin polling. The polling logic is common across the different flows, and the steps are described in detail in [Polling Okta](#polling-okta).
+
+### 4: Open Okta Verify
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/nodeexpress/enrollmentqrcodeintegrationsteps.md:76]: Why isn't there an Install Okta Verify section earlier on, like there is for Google Auth in the other guide? This shouldn't be a step in the enrollment flow.

File: packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/nodeexpress/enrollmentotherintegrationsteps.md [hunk 21]
```diff
@@ -0,0 +1,196 @@
+
+### 1: Initiate the sign-in and choose Okta Verify as the authenticator
+
+First, the user initiates the user sign-in with username and password and chooses Okta Verify as the authenticator to enroll in. These steps are shared between enrollment flows and described in [Initiate sign in for Okta Verify enrollment](#initiate-sign-in-for-okta-verify-enrollment).
+
+### 2: Display QR Code and option to select other channels
+
+After the user chooses Okta Verify (which is the last step in [Initiate sign in for Okta Verify enrollment](#initiate-sign-in-for-okta-verify-enrollment)), display the QR Code using the `IdxTransaction` response from `OktaAuth.idx.proceed()`. In addition, to the QR Code also display a link to choose an alternative enrollment channel (for example, email or SMS). The option to select a different enrollment channel is indicated by the `select-enrollment-channel` value under `Idxtransaction.availableSteps[n].name`.
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/nodeexpress/enrollmentotherintegrationsteps.md:8]:
```suggestion
Display the QR Code using the `IdxTransaction` response from `OktaAuth.idx.proceed()`. In addition, to the QR Code also display a link to choose an alternative enrollment channel (for example, email or SMS). The option to select a different enrollment channel is indicated by the `select-enrollment-channel` value under `Idxtransaction.availableSteps[n].name`.
```

**Review by danielmaharry-okta — CHANGES_REQUESTED** [review]
